### ISSUE TITLE
signInWithCredential failed: First argument "credential" must be a valid credential.

### ISSUE THREAD

**Issue by chrisdrohr** [firebase/firebaseui-web#367]
Hello i started getting this error after updating **firebaseui** and still after i switched to **firebaseui-web-react**.
can anyone point me in the right direction? Thankyou!

The error message shows after choosing an account below the loader

**Comment by bojeil-google** [comment]
Are you using the latest version of the library?

**Comment by chrisdrohr** [comment]
"firebase": "^4.12.1",
    "firebaseui": "^2.7.0",
 "react-firebaseui": "^1.2.3",

**Comment by chrisdrohr** [comment]
Sure, here you go https://dev.popcheckr.com/

**Comment by chrisdrohr** [comment]
Hi, is there any recommendations on what I can do to get it working in the meantime?

**Comment by wti806** [comment]
@chrisdrohr Can you try to downgrade to 2.6.x while we debugging it? I was not able to replicate this one with our own test app. Not sure if it's related to react environment.

**Comment by brentsuggs** [comment]
Did anything every come of this?  I am having the same issue but not with react.  I'm using webpack, foundation css, and just utilizing the email and password login option.

**Comment by chrisdrohr** [comment]
I got it to work again somehow by downgrading to 2.6.0, but i accidently updated to the latest version today and started experiencing the same issue. I spent the last few hours trying to get it working again by downgrading, changing my config, removing and installing all of my modules, and trying random versions of firebaseui but nothing worked.

I gave up and started implementing it [manually](https://firebase.google.com/docs/auth/web/google-signin) and it seems to working fine so far.

**Comment by wti806** [comment]
We still cannot reproduce this issue on our side, which makes it very hard to debug. One thing I suggest you guys do is to clone the repo and build the un-obfuscated library. This way you could help us to see where exactly the error happens.

**Comment by realgsong** [comment]
+1, I'm having same trouble with 2.7.0. Workaround is downgrading firebaseui to 2.6.1. Here's the npm list result grepped by 'firebase'

```
├─┬ firebase@4.6.2
│ ├─┬ @firebase/app@0.1.10
│ │ ├── @firebase/app-types@0.1.2
│ │ └── @firebase/util@0.1.10
│ ├── @firebase/auth@0.2.2
│ ├─┬ @firebase/database@0.1.11
│ │ ├── @firebase/database-types@0.1.2
│ ├─┬ @firebase/firestore@0.1.4
│ │ └── @firebase/webchannel-wrapper@0.2.8
│ ├─┬ @firebase/messaging@0.1.9
│ │ ├── @firebase/messaging-types@0.1.1
│ │ └── @firebase/util@0.1.8
│ ├─┬ @firebase/polyfill@0.1.6
│ ├─┬ @firebase/storage@0.1.9
│ │ └── @firebase/storage-types@0.1.3
├─┬ firebase-tools@3.18.4
│ ├─┬ firebase@2.4.2
├─┬ firebaseui@2.7.0
│ └─┬ firebase@4.13.1
│   ├─┬ @firebase/app@0.2.0
│   │ ├── @firebase/app-types@0.2.0
│   │ └── @firebase/util@0.1.11
│   ├─┬ @firebase/auth@0.4.2
│   │ └── @firebase/auth-types@0.2.1
│   ├─┬ @firebase/database@0.2.2
│   │ ├── @firebase/database-types@0.2.1
│   │ └── @firebase/logger@0.1.1
│   ├─┬ @firebase/firestore@0.4.1
│   │ ├── @firebase/firestore-types@0.3.0
│   ├─┬ @firebase/functions@0.1.1
│   │ ├── @firebase/functions-types@0.1.1
│   │ ├── @firebase/messaging-types@0.1.3
│   ├─┬ @firebase/messaging@0.2.4
│   │ └── @firebase/messaging-types@0.1.3
│   └─┬ @firebase/polyfill@0.3.1
```

**Comment by wti806** [comment]
@realgsong Could you provide more detail of what flow triggered the error? That would help us debug cause right now we are not able to replicate on our end. And also error message in console would help as well. Thank you so much!

**Comment by wti806** [comment]
@realgsong I noticed one thing. We installed firebase core sdk v4.6.2 on top level. And I assume you initialized the Auth instance with v4.6.2 and passed it to FirebaseUI. But FirebaseUI depends on 4.13.1 as you can see in the bottom of npm dependencies you post. Not sure if the version conflict could cause this problem. Can you try to use the same version of core sdk and try again? Thanks

**Comment by realgsong** [comment]
Updating firebase version to match dependency of firebaseui solved my case.
Thank you very much. :)

2018년 5월 5일 (토) 오전 6:28, wti806 <notifications@github.com>님이 작성:

**Comment by ackvf** [comment]
After updating to `firebase: "^5.1.0"`, to utilise cloud functions, every sign in method started to fail with the error `signInWithCredential failed: First argument "credential" must be a valid credential.`, except twitter with `Unable to determine the authorization endpoint for the specified provider. This may be an issue in the provider configuration.`

working versions:
    "firebase": "4.10.1",
    "firebaseui": "2.6.2",

not working after update:
    "firebase": "5.1.0",
    "firebaseui": "2.7.0",

**Comment by bojeil-google** [comment]
Are you saying updating to the following versions fixed the issue?
```
"firebase": "^5.2.0",
"firebaseui": "^3.1.1",
```

**Comment by ackvf** [comment]
Yes updating firebase and firebaseui fixed it, I didn't have to do any other change in the code.

**Comment by mancini0** [comment]
I'm encountering this error as well with firebase 5.3.0

    **"firebase": "5.3.0",
    "firebaseui": "^3.1.1",**
    "formik": "^0.11.11",
    "google-protobuf": "^3.6.0",
    "moment": "^2.22.2",
    "node-sass": "^4.9.0",
    "rc-slider": "^8.6.1",
    "react": "^16.4.0",
    "react-dom": "^16.4.0",
    **"react-firebaseui": "^1.2.3",**
